addons/purchase/tests/test_access_rights.py

```diff
@@ -12,18 +12,28 @@ class TestPurchaseInvoice(AccountTestInvoicingCommon):
     def setUpClass(cls, chart_template_ref=None):
         super().setUpClass(chart_template_ref=chart_template_ref)
 
+        cls.company = cls.company_data["company"]
+
         # Create a users
         group_purchase_user = cls.env.ref('purchase.group_purchase_user')
         group_employee = cls.env.ref('base.group_user')
         group_partner_manager = cls.env.ref('base.group_partner_manager')
+        cls.group_purchase_manager = cls.env.ref('purchase.group_purchase_manager')
+        user_admin = cls.env.ref("base.user_admin")
 
         cls.purchase_user = cls.env['res.users'].with_context(
             no_reset_password=True
         ).create({
             'name': 'Purchase user',
             'login': 'purchaseUser',
             'email': 'pu@odoo.com',
-            'groups_id': [(6, 0, [group_purchase_user.id, group_employee.id, group_partner_manager.id])],
+            'company_ids': [(4, cls.company.id)],
+            'company_id': cls.company.id,
```

Comment:
As we set the user's company only for `test_double_validation`, I would be more confident if we don't set it here